src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-form/crud-form.component.ts

```diff
@@ -32,16 +38,29 @@ export class CrudFormComponent implements OnInit {
     private dataGatewayService: DataGatewayService,
     private activatedRoute: ActivatedRoute,
     private taskWrapper: TaskWrapperService,
-    private location: Location
+    private location: Location,
+    private formlyJsonschema: FormlyJsonschema
   ) {}
 
   ngOnInit(): void {
+    this.action = 'Create';
     this.activatedRoute.data.subscribe((data: any) => {
       this.resource = data.resource;
       this.dataGatewayService.list(`ui-${this.resource}`).subscribe((response: any) => {
         this.title = response.forms[0].control_schema.title;
         this.uiSchema = response.forms[0].ui_schema;
-        this.controlSchema = response.forms[0].control_schema;
+        let xd = response.forms[0].control_schema;
+        this.controlSchema = this.formlyJsonschema.toFieldConfig(xd).fieldGroup;
+        for (let i = 0; i < this.controlSchema.length; i++) {
+          for (let j = 0; j < this.uiSchema.length; j++) {
+            if (this.controlSchema[i].key == this.uiSchema[j].key) {
+              this.controlSchema[i].className = this.uiSchema[j].className;
+              this.controlSchema[i].props.templateOptions = this.uiSchema[j].templateOptions;
+              this.controlSchema[i].props.required = this.uiSchema.required;
```

Comment:
When I was reading the Python code in this PR it felt tad clunky to see CSS classes and HTML-like stuff over there. What about having this architecture:
* Python backend (CRUD) generates the 'abstract' layout schema. It doesn't know about classes or icons, or Form frameworks, just about generic (JSONSchema-like) data structures, types, and connections between those.
* Angular front-end component: this CRUD-form component should just consume/is fed by the output of a(n adapter) service and nothing else.
* Angular CRUD-Form adapter service: it takes the abstract data structure schema from the Python backend and generates the required schema (Formly) with all the HTML/CSS decorations, and all the inputs required to instantiate the CRUD Form component.

This is a classic design pattern.

Reply:
I created a `crud-form-adapter.service` and moved the data processing logic to that service. Now the `OnInit()` will only have the subscriber.